shopfloor/services/checkout.py

```diff
@@ -83,19 +83,28 @@ def _response_for_manual_selection(self, message=None):
         return self._response(next_state="manual_selection", data=data, message=message)
 
     def _response_for_select_package(self, picking, lines, message=None):
+        with_pack, wo_pack = self._get_allow_package_options()
         return self._response(
             next_state="select_package",
             data={
                 "selected_move_lines": self._data_for_move_lines(lines.sorted()),
                 "picking": self.data.picking(picking),
                 "packing_info": self._data_for_packing_info(picking),
-                "no_package_enabled": not self.options.get(
-                    "checkout__disable_no_package"
-                ),
+                "allow_with_package": with_pack,
+                "allow_without_package": wo_pack,
             },
             message=message,
         )
 
+    def _get_allow_package_options(self):
+        if not self.work.menu.package_process_type:
+            return True, True
+
+        if self.work.menu.package_process_type == "with_package":
+            return True, False
+
+        return False, True
+
```

Comment:
I think we should split this into two separate methods. This would simplify the code later. 

```suggestion
    def _is_process_with_package_enabled(self):
        if not self.work.menu.package_process_type:
            return True
        if self.work.menu.package_process_type == "with_package":
            return True
         return False
         
     def _is_process_without_package_enabled(self):
         if not self.work.menu.package_process_type:
             return True
         if self.work.menu.package_process_type == "without_package":
             return True
         return False

```